concordium-std-derive/src/lib.rs

```diff
@@ -1144,7 +1150,20 @@ fn schema_type_derive_worker(input: TokenStream) -> syn::Result<TokenStream> {
                 .variants
                 .iter()
                 .map(|variant| {
-                    let variant_name = &variant.ident.to_string();
+                    let variant_name = if let Some(renamed_variant_name) =
+                        find_field_attribute_value(&variant.attrs, "rename")?
+                    {
+                        match renamed_variant_name {
+                            syn::Lit::Str(renamed_variant_name) => renamed_variant_name.value(),
+                            _ => return Err(syn::Error::new(
+                                renamed_variant_name.span(),
+                                "Rename attribute value must be a string.",
+                            )),
+                        }
+                    } else {
+                        variant.ident.to_string()
+                    };
+
```

Comment:
We should also check for name collisions, which I imagine will be possible now.

Reply:
Ah yes :) Done.